knora/dsplib/utils/xml_upload.py

```diff
@@ -274,6 +275,32 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
     permissions_lookup: dict[str, Permissions] = {s: perm.get_permission_instance() for s, perm in permissions.items()}
     resclass_name_2_type: dict[str, type] = {s: res_inst_factory.get_resclass_type(s) for s in res_inst_factory.get_resclass_names()}
 
+    # check if the data in the XML is consistent with the ontology
+    if verbose:
+        print("Check if the resource types and properties in your XML are consistent with the ontology...")
+    knora_properties = resclass_name_2_type[resources[0].restype].knora_properties
+    for resource in resources:
+        if resource.restype not in resclass_name_2_type:
+            print(f"=========================\n"
+                  f"ERROR: Resource '{resource.label}' (ID: {resource.id}) has an invalid resource type "
+                  f"'{resource.restype}'. Is your syntax correct? Remember the rules:\n"
+                  f" - DSP-API internals: <resource restype=\"restype\">         (will be interpreted as 'knora-api:restype')\n"
+                  f" - current onto:      <resource restype=\":restype\">        ('restype' must be defined in the 'resources' section of your ontology)\n"
+                  f" - other onto:        <resource restype=\"other:restype\">   (not yet implemented: 'other' must be defined in the same JSON project file than your ontology)")
```

Comment:
```suggestion
                  f" - other ontology:        <resource restype=\"other:restype\">   (not yet implemented: 'other' must be defined in the same JSON project file than your ontology)")
```

Reply:
same everywhere: I would use "ontology" instead of "onto" which sounds rather informal to me.